meshroom/nodes/aliceVision/SketchfabUpload.py

```diff
@@ -184,27 +169,27 @@ class SketchfabUpload(desc.Node):
             uid=[],
         ),
     ]
-    
-    def upload(self, apiToken, modelFile, data, chunk):
+
+    def upload(self, apiToken, modelFile, data, logger):
         modelEndpoint = 'https://api.sketchfab.com/v3/models'
         f = open(modelFile, 'rb')
         file = {'modelFile': (os.path.basename(modelFile), f.read())}
         file.update(data)
         f.close()
         (files, contentType) = requests.packages.urllib3.filepost.encode_multipart_formdata(file)
         headers = {'Authorization': 'Token {}'.format(apiToken), 'Content-Type': contentType}
-        body = BufferReader(files, progressUpdate, cb_kwargs={'logManager': chunk.logManager}, stopped=self.stopped)
-        chunk.logger.info('Uploading...')
+        body = BufferReader(files, logger, self.stopped)
+        logger.info('Uploading...')
         try:
             r = requests.post(
                 modelEndpoint, **{'data': body, 'headers': headers})
-            chunk.logManager.completeProgressBar()
         except requests.exceptions.RequestException as e:
-            chunk.logger.error(u'An error occured: {}'.format(e))
-            raise RuntimeError() 
+            raise RuntimeError(u'An error occured: {}'.format(e))
+        finally:
+            os.remove(modelFile)
```

Comment:
The upoload function should not have the responsability of removing the input file.
The creation/removal of the temp file should be at the same level.